### PR TITLE
[PLATFORM-279] Stream editor: Configure

### PR DESCRIPTION
**Pull request by tumppi**
Add configure section for stream editor. Fields can be modified and added. Additional settings like autoconfigure and signing are not currently saved because there is no backend support.

This also contains changes from https://github.com/streamr-dev/streamr-platform/pull/188.

### CODE REVIEW

File: app/src/userpages/components/ProfilePage/profilePage.pcss [hunk 1]
```diff
@@ -7,6 +7,7 @@
   z-index: 11;
   top: -4em;
   position: relative;
+  font-size: 14px;
```
Comment (juhah) [inline app/src/userpages/components/ProfilePage/profilePage.pcss]: Use `rem` here?

File: app/src/userpages/components/StreamPage/Show/ConfigureView/NewFieldEditor/index.jsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+    validate = () => {
+        const { name } = this.state
+        const { previousFields } = this.props
+        let error = null
+
+        if (name.length === 0) {
+            error = I18n.t('userpages.streams.edit.configure.newFieldEditor.error.emptyName')
+        }
+        if (previousFields.find((field) => field.name === name)) {
```
Comment (juhah) [inline app/src/userpages/components/StreamPage/Show/ConfigureView/NewFieldEditor/index.jsx]: `error` is overwritten here if both conditions are true.

File: app/src/userpages/modules/userPageStreams/actions.js [hunk 3]
```diff
@@ -320,9 +321,9 @@ export const createStream = (options: { name: string, description: ?string }) =>
 export const updateStream = (stream: Stream) => (dispatch: Function) => {
     dispatch(updateStreamRequest())
     return services.putStream(stream.id, stream)
-        .then(handleEntities(streamSchema, dispatch))
-        .then((id) => {
-            dispatch(updateStreamSuccess(id))
+        .then(() => handleEntities(streamSchema, dispatch)(stream))
+        .then(() => {
+            dispatch(updateStreamSuccess(stream))
```
Comment (juhah) [inline app/src/userpages/modules/userPageStreams/actions.js]: You could have also changed the parameter for `updateStreamSuccess` to an id, it's not really used for anything.

File: app/src/userpages/modules/userPageStreams/services.js [hunk 4]
```diff
@@ -10,7 +10,7 @@ export const getStream = (id: StreamId): ApiResult<Stream> => get(formatApiUrl('
 
 export const postStream = (stream: NewStream): ApiResult<Stream> => post(formatApiUrl('streams'), stream)
 
-export const putStream = (id: StreamId, stream: Stream): ApiResult<Stream> => put(formatApiUrl('streams', id), stream)
+export const putStream = (id: StreamId, stream: Stream): ApiResult<null> => put(formatApiUrl('streams', id), stream)
```
Comment (juhah) [inline app/src/userpages/modules/userPageStreams/services.js]: Weird that this endpoint behaves differently 😲

File: app/src/userpages/components/StreamPage/constants.js [hunk 5]
```diff
@@ -1,12 +1,36 @@
...
+export const fieldTypes = ['number', 'boolean', 'map', 'list', 'string']
+
+export const fieldValidators = {
+    number: (f: string): boolean => !Number.isNaN(parseFloat(f)),
+    string: (f: string): boolean => f instanceof String || typeof f === 'string',
+    boolean: (f: string): boolean => f === 'true' || f === 'false',
+    map: (f: string): boolean => {
+        try {
+            const m = JSON.parse(f)
+            return typeof m === 'object' && m !== null
```
Comment (mondoreale) [inline app/src/userpages/components/StreamPage/constants.js]: `typeof []` gives `object` too, FYI.

File: app/src/userpages/components/ProfilePage/profilePage.pcss (hunk 1 above)
Comment (tumppi) [reply app/src/userpages/components/ProfilePage/profilePage.pcss]: Hmm, this is coming from https://github.com/streamr-dev/streamr-platform/pull/188. Do we actually need this?

File: app/src/userpages/components/StreamPage/Show/ConfigureView/NewFieldEditor/index.jsx (hunk 2 above)
Comment (tumppi) [reply app/src/userpages/components/StreamPage/Show/ConfigureView/NewFieldEditor/index.jsx]: Yes, field.name should never be empty. If it is, then something is wrong with the backend. Also I think it's correct behaviour to complain about duplicate field name if this happens :)

File: app/src/userpages/components/StreamPage/constants.js (hunk 5 above)
Comment (tumppi) [reply app/src/userpages/components/StreamPage/constants.js]: Ah, I just moved it to this file. I actually see it's not used anywhere so I might as well remove it.

**Pull request merged**